dds/DCPS/DiscoveryBase.h

```diff
@@ -526,13 +520,35 @@ namespace OpenDDS {
         return rid;
       }
 
+#ifdef OPENDDS_SECURITY
+      void cleanup_secure_writer(const RepoId& publicationId)
+      {
+        using namespace DDS::Security;
+
+        const DatawriterCryptoHandle dwch = get_handle_registry()->get_local_datawriter_crypto_handle(publicationId);
+        SecurityException ex = {"", 0, 0};
+        if (!get_crypto_key_factory()->unregister_datawriter(dwch, ex)) {
+          ACE_ERROR((LM_ERROR,
```

Comment:
Dependent on some log level?

Reply:
Since it's security related, I will change it to use a `security_debug` flag.